domain/schema/model/sql/0020-network.sql

```diff
@@ -14,9 +14,295 @@ CREATE UNIQUE INDEX idx_cloud_service_application
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_network_address_scope_name
+ON network_address_scope (name);
+
+INSERT INTO network_address_scope VALUES
+(0, 'unknown'),
+(1, 'public'),
+(2, 'local-cloud'),
+(3, 'local-machine'),
+(4, 'link-local');
+
+-- ip_address_config_type defines valid network
+-- link configuration types.
+CREATE TABLE ip_address_config_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_config_type_name
+ON ip_address_config_type (name);
+
+INSERT INTO ip_address_config_type VALUES
+(0, 'unknown'),
+(1, 'dhcp'),
+(2, 'static'),
+(3, 'manual'),
+(4, 'loopback');
+
+CREATE TABLE ip_address (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    -- The value of the configured IP address.
+    -- e.g. 192.168.1.2 or 2001:db8::1.
+    address_value TEXT NOT NULL,
+    -- one of ipv4, ipv6 etc.
+    type_id INT NOT NULL,
+    -- one of dhcp, static, manual, loopback etc.
+    config_type_id INT NOT NULL,
+    -- one of host, provider
+    origin_id INT NOT NULL,
+    -- one of public, local-cloud, local-machine, link-local etc.
+    scope_id INT NOT NULL,
+    -- the link layer device this address belongs to.
+    device_uuid TEXT NOT NULL,
+
+    -- indicates that this address is not the primary.
+    -- address associated with the NIC.
+    is_secondary BOOLEAN DEFAULT false,
+
+    -- indicates whether this address is a virtual/floating/shadow.
+    -- address assigned to a NIC by a provider rather than being
+    -- associated directly with a device on-machine.
+    is_shadow BOOLEAN DEFAULT false,
+
+    CONSTRAINT fk_ip_address_link_layer_device
+    FOREIGN KEY (device_uuid)
+    REFERENCES link_layer_device (uuid),
+    CONSTRAINT fk_ip_address_origin
+    FOREIGN KEY (origin_id)
+    REFERENCES ip_address_origin (id),
+    CONSTRAINT fk_ip_address_type
+    FOREIGN KEY (type_id)
+    REFERENCES ip_address_type (id),
+    CONSTRAINT fk_ip_address_config_type
+    FOREIGN KEY (config_type_id)
+    REFERENCES ip_address_config_type (id),
+    CONSTRAINT fk_ip_address_scope
+    FOREIGN KEY (scope_id)
+    REFERENCES network_address_scope (id)
+);
+
+CREATE TABLE ip_address_dns_server_address (
+    address_uuid TEXT NOT NULL,
+    dns_server_address_uuid TEXT NOT NULL,
+    CONSTRAINT fk_ip_address_dns_server_address_dns_server_address
+    FOREIGN KEY (dns_server_address_uuid)
+    REFERENCES ip_address (uuid),
+    CONSTRAINT fk_ip_address_dns_server_address_ip_address
+    FOREIGN KEY (address_uuid)
+    REFERENCES ip_address (uuid),
+    PRIMARY KEY (dns_server_address_uuid, address_uuid)
+);
+
+CREATE TABLE ip_address_dns_search_domain (
+    address_uuid TEXT NOT NULL,
+    search_domain TEXT NOT NULL,
+    CONSTRAINT fk_ip_address_dns_search_domain_ip_address
+    FOREIGN KEY (address_uuid)
+    REFERENCES ip_address (uuid),
+    PRIMARY KEY (address_uuid, search_domain)
+);
+
+CREATE TABLE fqdn_address (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    hostname TEXT NOT NULL,
```

Comment:
I am being a nitty nitter here. Hostname is only one portion of an fqdn?

Are we modelling the fqdn or the hostname + the dns name?

The names and data is in conflict with each other for this table.

Reply:
We'll use 2 tables - one for hostname and one for fqdn